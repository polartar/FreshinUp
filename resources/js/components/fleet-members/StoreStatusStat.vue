<template>
  <v-layout
    column
    class="pa-4"
    justify-center
    align-center
  >
    <v-flex
      class="display-3"
      :class="color + '--text'"
    >
      {{ value }}
    </v-flex>
    <v-flex class="grey--text text-uppercase font-weight-bold">
      {{ label }}
    </v-flex>
  </v-layout>
</template>

<script>
export default {
  props: {
    label: { type: String, default: '' },
    value: { type: Number, default: 0 },
    color: { type: String, default: 'grey' }
  }
}
</script>
