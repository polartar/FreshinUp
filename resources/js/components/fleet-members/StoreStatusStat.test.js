import { mount } from '@vue/test-utils'
import Component from './StoreStatusStat.vue'
import * as Stories from './StoreStatusStat.stories'
import { FIXTURE_STORE_STATUS_STATS } from '../../../../tests/Javascript/__data__/storeStatusStats'

const item = FIXTURE_STORE_STATUS_STATS[0]
describe('components/fleet-members/StoreStatusStat', () => {
  describe('Snapshots', () => {
    test('Default', () => {
      const wrapper = mount(Stories.Default())
      expect(wrapper.element).toMatchSnapshot()
    })
    test('Populated', () => {
      const wrapper = mount(Stories.Populated())
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Props & Computed', () => {
    test('label', async () => {
      const wrapper = mount(Component)
      expect(wrapper.vm.label).toEqual('')

      wrapper.setProps({
        label: item.label
      })
      await wrapper.vm.$nextTick()
      expect(wrapper.vm.label).toBe(item.label)
    })
    test('color', async () => {
      const wrapper = mount(Component)
      expect(wrapper.vm.color).toEqual('grey')

      wrapper.setProps({
        color: item.color
      })
      await wrapper.vm.$nextTick()
      expect(wrapper.vm.color).toBe(item.color)
    })
    test('value', async () => {
      const wrapper = mount(Component)
      expect(wrapper.vm.value).toEqual(0)

      wrapper.setProps({
        value: item.value
      })
      await wrapper.vm.$nextTick()
      expect(wrapper.vm.value).toBe(item.value)
    })
  })
})
