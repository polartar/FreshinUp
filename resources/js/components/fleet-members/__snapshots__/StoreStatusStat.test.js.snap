// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/fleet-members/StoreStatusStat Snapshots Default 1`] = `
<div
  class="container"
>
  <div
    class="layout pa-4 column justify-center align-center"
  >
    <div
      class="flex display-3 grey--text"
    >
      
    0
  
    </div>
     
    <div
      class="flex grey--text text-uppercase font-weight-bold"
    >
      
    
  
    </div>
  </div>
</div>
`;

exports[`components/fleet-members/StoreStatusStat Snapshots Populated 1`] = `
<div
  class="container"
>
  <div
    class="layout pa-4 column justify-center align-center"
  >
    <div
      class="flex display-3 warning--text"
    >
      
    24
  
    </div>
     
    <div
      class="flex grey--text text-uppercase font-weight-bold"
    >
      
    Pending
  
    </div>
  </div>
</div>
`;
